sys/include/net/sock/ip.h

```diff
@@ -0,0 +1,204 @@
+ * @return  -ENOTCONN, when @p sock has no remote end point bound to it.
+ */
+int sock_ip_get_remote(sock_ip_t *sock, sock_ip_ep_t *ep);
+
+/**
+ * @brief   Receives a message over IPv4/IPv6 from remote end point
+ *
+ * @pre `(sock != NULL) && (data != NULL) && (max_len > 0)`
+ *
+ * @param[in] sock      A raw IPv4/IPv6 sock object.
+ * @param[out] data     Pointer where the received data should be stored.
+ * @param[in] max_len   Maximum space available at @p data.
+ *                      If received data exceeds @p max_len the data is
+ *                      truncated and the remaining data can be retrieved
+ *                      later on.
+ * @param[in] timeout   Timeout for receive in microseconds.
+ *                      This value can be ignored (no timeout) if the
+ *                      @ref sys_xtimer module is not present and the stack does
+ *                      not support timeouts on its own.
+ *                      May be 0 for no timeout.
+ * @param[out] remote   Remote end point of the received data.
+ *                      May be NULL, if it is not required by the application.
+ *
+ * @note    Function blocks if no packet is currently waiting.
+ *
+ * @return  The number of bytes received on success.
+ * @return  0, if no received data is available, but everything is in order.
+ * @return  -EADDRNOTAVAIL, if local of @p sock is not given.
+ * @return  -ENOBUFS, if buffer space is not large enough to store received
+ *          data.
+ * @return  -ENOMEM, if no memory was available to receive @p data.
+ * @return  -EPROTO, if source address of received packet did not equal
+ *          the remote of @p sock.
+ * @return  -ETIMEDOUT, if @p timeout expired.
+ */
+ssize_t sock_ip_recv(sock_ip_t *sock, void *data, size_t max_len,
+                     uint32_t timeout, sock_ip_ep_t *remote);
+
+/**
+ * @brief   Sends a message over IPv4/IPv6 to remote end point
+ *
+ * @pre `((sock != NULL || remote != NULL)) && (if (len != 0): (data != NULL))`
+ *
+ * @param[in] sock      A raw IPv4/IPv6 sock object. May be NULL.
+ *                      A sensible local end point should be selected by the
+ *                      stack in that case.
+ * @param[in] data      Pointer where the received data should be stored.
+ *                      May be `NULL` if `len == 0`.
+ * @param[in] len       Maximum space available at @p data.
+ * @param[in] proto     Protocol to use in the packet send, in case
+ *                      `sock == NULL`. If `sock != NULL` this parameter will be
+ *                      ignored.
+ * @param[in] remote    Remote end point for the send data.
+ *                      May be `NULL`, if @p sock has a remote end point.
+ *                      sock_ip_ep_t::family may be AF_UNSPEC, if local
+ *                      end point of @p sock provides this information.
+ *
+ * @note    Function blocks until packet is handed to the stack.
+ *
+ * @return  The number of bytes send on success.
```

Comment:
If anyone is going to fix this: there are various more occurrences of this typo all over the place.